src/gt4py/next/program_processors/runners/dace_iterator/itir_to_sdfg.py

```diff
@@ -189,6 +197,103 @@ def add_storage(
             raise NotImplementedError()
         self.storage_types[name] = type_
 
+    def generate_temporaries(
+        self, node_params: list[Sym], defs_state: dace.SDFGState, program_sdfg: dace.SDFG
+    ) -> dict[str, TaskletExpr]:
+        """Create a table of symbols which are used to define array shape, stride and offset for temporaries."""
+        symbol_map: dict[str, TaskletExpr] = {}
+        # The shape of temporary arrays might be defined based on the shape of other input/output fields.
+        # Therefore, here we collect the symbols used to define data-field parameters that are not temporaries.
+        for sym in node_params:
+            if all([sym.id != tmp.id for tmp in self.tmps]) and sym.kind != "Iterator":
```

Comment:
This seems very indirect to me, and not that reliable, because what is with a scalar input, such as the timestep, wouldn't that be matched here as well.
Since all strides/shapes are generated by one function, thy all have a predicable name, which you could match using regex.